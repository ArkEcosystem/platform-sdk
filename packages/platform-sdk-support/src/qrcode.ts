import QRious from "qrious";

export class QRCode {
	readonly #instance: QRious;

	private constructor() {
		this.#instance = new QRious();
	}

	public static fromString(value: string): QRCode {
		return new QRCode().set({ value });
	}

	public static fromObject(value: object): QRCode {
		return new QRCode().set({
			value: JSON.stringify(value),
		});
	}

	public set(value: object): QRCode {
		this.#instance.set(value);

		return this;
	}

	public toDataURL(): string {
		return this.#instance.toDataURL("image/png");
	}
}
