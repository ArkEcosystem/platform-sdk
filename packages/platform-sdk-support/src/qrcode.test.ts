import "jest-extended";

import { QRCode } from "./qrcode";

test("#fromString", () => {
	expect(QRCode.fromString("https://google.com")).toBeInstanceOf(QRCode);
});

test("#fromObject", () => {
	expect(QRCode.fromObject({ url: "https://google.com" })).toBeInstanceOf(QRCode);
});

test("#toDataURL", async () => {
	const actual: string = await QRCode.fromString("https://google.com").toDataURL();

	expect(actual).toStartWith("data:image/png;base64,");
	expect(actual).toMatchSnapshot();
});

test("#toDataURL with options", async () => {
	const actual: string = await QRCode.fromString("https://google.com").toDataURL({ width: 250, margin: 0 });

	expect(actual).toStartWith("data:image/png;base64,");
	expect(actual).toMatchSnapshot();
});

describe.each(["utf8", "svg", "terminal"])("%s", (type) => {
	it("should turn into a data URL", async () => {
		await expect(QRCode.fromString("https://google.com").toDataURL()).resolves.toMatchSnapshot();
	});

	it("should turn into a string", async () => {
		await expect(QRCode.fromString("https://google.com").toString(type)).resolves.toMatchSnapshot();
	});
});

it("should turn into a utf-8 string if no argument is given", async () => {
	await expect(QRCode.fromString("https://google.com").toString()).resolves.toMatchSnapshot();
});
