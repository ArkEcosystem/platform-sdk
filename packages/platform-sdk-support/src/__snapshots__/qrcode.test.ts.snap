// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toDataURL 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAIQAAACECAYAAABRRIOnAAAAAklEQVR4AewaftIAAAOKSURBVO3BQY5jiRUDwcwH3f/KdC+84OoDglTlmTYjzB/M/NcxU46ZcsyUY6YcM+WYKcdMOWbKMVOOmXLMlGOmHDPlmCnHTHnxIZXflIRPqLwjCU9UWhKaym9KwieOmXLMlGOmvPiyJHyTyjtU3pGEJyrflIRvUvmmY6YcM+WYKS9+mMo7kvAOlSdJaCqfSEJT+YTKO5Lwk46ZcsyUY6a8+Mskoak8UWlJaEl4koS/yTFTjplyzJQXf7kkNJUnKi0JT1RaEv7NjplyzJRjprz4YUn4TSrvSEJTaSo/KQn/JMdMOWbKMVNefJnK/1ISmkpLQlNpSWgqLQlN5R0q/2THTDlmyjFTzB/8i6l8IglNpSXh/8kxU46ZcswU8wcfUGlJaCrflITfpPIkCU9UvikJP+mYKcdMOWaK+YMvUnmShE+otCQ0lSdJ+CaVloQnKi0J71BpSfimY6YcM+WYKS9+mUpLwhOVloSm8ptUWhKaSkvCE5UnSXii0pLwiWOmHDPlmCkvviwJTaUloak8ScI3qTxJQlNpSWgqn0jCO5LQVL7pmCnHTDlmivmDD6g8SUJT+UQSmkpLQlNpSfhJKk+S0FRaEprKO5LwiWOmHDPlmCkvviwJTaUloak8SUJTaUn4hMo7kvAkCU9UWhKaSktCU2lJ+KZjphwz5ZgpL35YEj6h0pLQVD6RhKbyROUdSXii8gmVloRPHDPlmCnHTHnxy1RaEppKS0JTaUl4ovJE5UkSmspPSsKTJDSVbzpmyjFTjpny4oeptCQ0lScqLQlN5ZuS0FQ+ofIJlZaEloRvOmbKMVOOmWL+4F9M5R1J+ITKkyS8Q+VJEn7TMVOOmXLMlBcfUvlNSWhJeIfKkyQ8SUJTeaLSkvAkCU3lHUn4xDFTjplyzJQXX5aEb1J5otKS8I4kPFFpSXhHEt6h0pLwm46ZcsyUY6a8+GEq70jCP0kSmsoTlU8koam0JDSVloRPHDPlmCnHTHnxl1NpSWgqn0hCU3mShKbSVJ6o/KRjphwz5ZgpL/4yKi0JT5LQVJ4koam0JLwjCU3lSRKayjcdM+WYKcdMefHDkvCTkvAJlSdJaCpPVL4pCb/pmCnHTDlmyosvU/lNKu9IwpMkNJWWhCcqLQlN5R0qv+mYKcdMOWaK+YOZ/zpmyjFTjplyzJRjphwz5Zgpx0w5ZsoxU46ZcsyUY6YcM+WYKf8B7SOFJl3T5SkAAAAASUVORK5CYII="`;

exports[`#toDataURL with options 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAPoAAAD6CAYAAACI7Fo9AAAAAklEQVR4AewaftIAAAWDSURBVO3BQW4kBwwEwSxi/v/l8t51IRZt95jKiABFP7TlmyXhDW15UhI22rKRBP00SDpvkHTeIOm8QdJ5g6TzBknnDZLOGySdN0g678NSWy5IwhuS8Ia2PCkJF7TlgiRsDJLOGySdN0g6b5B03iDpvEHSeYOk8wZJ5w2SzvvwsCS8oS1vSMKT2rKRhG/Wlo0kfLMkvKEtTxoknTdIOm+QdN4g6bxB0nmDpPMGSecNks4bJJ33Qf+Jtmwk4UlJ2GjLRlue1Bb9+wZJ5w2SzhsknTdIOm+QdN4g6bxB0nmDpPMGSed90Fdpy0YSnpSEjbY8KQkbbdHfGySdN0g6b5B03iDpvEHSeYOk8wZJ5w2SzhsknffhYW3RT0l4Q1s2krCRhN+kLRcMks4bJJ03SDpvkHTeIOm8QdJ5g6TzBknnDZLO+7CUBP29tmwkYaMtG0nYaMtGEjbaspGENyThNxkknTdIOm+QdN4g6bxB0nmDpPMGSecNks4bJJ2X/oH+WhK+WVs2krDRFv3/DJLOGySdN0g6b5B03iDpvEHSeYOk8wZJ5w2SzgtQFtqykYQL2qKfkvCktjwpCRe05UmDpPMGSecNks4bJJ03SDpvkHTeIOm8QdJ5g6Tz0j9YSMKT2vLNkrDRlo0kPKktFyRhoy1PSsJGW96QhI22bAySzhsknTdIOm+QdN4g6bxB0nmDpPMGSecNks778JIkbLTlSUnYaMtGEvRTEjbaspGEjbY8KQlPasuTkrAxSDpvkHTeIOm8QdJ5g6TzBknnDZLOGySdN0g678NSWzaSsNGWjSQ8qS0XJOFJbdlIwkZbNpLwzdryhrZsJGFjkHTeIOm8QdJ5g6TzBknnDZLOGySdN0g6b5B0XoDyoLZsJOGbtWUjCRtt2UjCRlt+kyQ8qS0bSdhoy0YS3jBIOm+QdN4g6bxB0nmDpPMGSecNks4bJJ03SDrvw1JbNpKw0ZaNJDypLRtJ2GjLN0vCG9rypLY8KQkbbdlIwkZbNpKw0ZaNQdJ5g6TzBknnDZLOGySdN0g6b5B03iDpvEHSeR8e1pZvloSNtmwk4Zu1ZSMJT0rCG9rypCR8syRsDJLOGySdN0g6b5B03iDpvEHSeYOk8wZJ5w2SzvvwkiRstGUjCRtt2UjCRluelIQnJeFJbdlIwm/Slie1ZSMJG4Ok8wZJ5w2SzhsknTdIOm+QdN4g6bxB0nmDpPM+PCwJG23ZSMKTkrDRlo0kXNCWjSR8syR8syRstGWjLRuDpPMGSecNks4bJJ03SDpvkHTeIOm8QdJ5g6Tz0j/QX0vCG9ryzZLwpLa8IQlPassbBknnDZLOGySdN0g6b5B03iDpvEHSeYOk8wZJ5wUo+qEt3ywJT2rLG5Kw0ZaNJGy0ZSMJG23ZSMIbBknnDZLOGySdN0g6b5B03iDpvEHSeYOk8wZJ531YassFSXhSEjba8oa2PCkJG215Q1vekISNtrxhkHTeIOm8QdJ5g6TzBknnDZLOGySdN0g6b5B03oeHJeENbdG/ry0bSXhSEr5ZWzaSsNGWjSRsDJLOGySdN0g6b5B03iDpvEHSeYOk8wZJ5w2Szvugr5KEjbZsJOGbtWUjCU9qy0YSNpLwpCQ8aZB03iDpvEHSeYOk8wZJ5w2SzhsknTdIOm+QdN4H/SeSsNGWJ7VlIwlPastGEjba8oa2bCThSW3ZSMLGIOm8QdJ5g6TzBknnDZLOGySdN0g6b5B03iDpvA8Pa8tv0pZvloQntWUjCU9KwgVtecMg6bxB0nmDpPMGSecNks4bJJ03SDpvkHTeIOm8D0tJ0E9JeENbntSWjSRstOVJSdhoy0YS3pCENwySzhsknTdIOm+QdN4g6bxB0nmDpPMGSecNks77B3cqZSaEd3FJAAAAAElFTkSuQmCC"`;

exports[`svg should turn into a string 1`] = `
"<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 33 33\\" shape-rendering=\\"crispEdges\\"><path fill=\\"#ffffff\\" d=\\"M0 0h33v33H0z\\"/><path stroke=\\"#000000\\" d=\\"M4 4.5h7m4 0h3m2 0h1m1 0h7M4 5.5h1m5 0h1m2 0h3m2 0h3m1 0h1m5 0h1M4 6.5h1m1 0h3m1 0h1m1 0h2m1 0h1m2 0h1m1 0h1m1 0h1m1 0h3m1 0h1M4 7.5h1m1 0h3m1 0h1m1 0h1m1 0h3m5 0h1m1 0h3m1 0h1M4 8.5h1m1 0h3m1 0h1m1 0h1m2 0h3m2 0h1m1 0h1m1 0h3m1 0h1M4 9.5h1m5 0h1m1 0h4m1 0h4m1 0h1m5 0h1M4 10.5h7m1 0h1m1 0h1m1 0h1m1 0h1m1 0h1m1 0h7M12 11.5h4m1 0h1M4 12.5h1m1 0h5m7 0h2m2 0h5M5 13.5h2m4 0h1m1 0h2m5 0h1m2 0h1m3 0h1M5 14.5h3m2 0h1m1 0h3m4 0h1m1 0h1m1 0h3m1 0h2M6 15.5h1m1 0h2m2 0h5m1 0h1m1 0h2m1 0h1m4 0h1M4 16.5h2m1 0h4m1 0h1m1 0h1m6 0h2m1 0h1m1 0h3M4 17.5h1m1 0h1m1 0h2m1 0h1m2 0h1m1 0h3m4 0h1m1 0h1m1 0h1M4 18.5h1m3 0h3m1 0h3m2 0h5m1 0h3m1 0h2M4 19.5h1m2 0h1m1 0h1m1 0h1m2 0h1m1 0h2m1 0h3m1 0h2m3 0h1M4 20.5h1m1 0h1m1 0h3m1 0h3m1 0h1m1 0h7m1 0h1M12 21.5h3m4 0h2m3 0h2M4 22.5h7m4 0h2m3 0h1m1 0h1m1 0h1m1 0h3M4 23.5h1m5 0h1m1 0h1m2 0h1m2 0h1m1 0h1m3 0h2M4 24.5h1m1 0h3m1 0h1m1 0h1m2 0h1m3 0h6m1 0h1m1 0h1M4 25.5h1m1 0h3m1 0h1m1 0h1m1 0h1m1 0h4m1 0h2m1 0h5M4 26.5h1m1 0h3m1 0h1m1 0h2m3 0h2m1 0h1m4 0h2m1 0h1M4 27.5h1m5 0h1m5 0h2m1 0h7m2 0h1M4 28.5h7m1 0h3m1 0h1m1 0h1m2 0h1m1 0h6\\"/></svg>
"
`;

exports[`terminal should turn into a string 1`] = `
"[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m
[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m
[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m
[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m
[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m
[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m
[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[47m  [0m[47m  [0m[40m  [0m[47m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[40m  [0m[47m  [0m
[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m[47m  [0m
"
`;

exports[`utf8 should turn into a string 1`] = `
"                                 
                                 
    █▀▀▀▀▀█  ▄▄█▀▀▄▄█ █▀▀▀▀▀█    
    █ ███ █ █▀▄█▄ ▀ ▀ █ ███ █    
    █ ▀▀▀ █ █▄▄█▀█▄▄█ █ ▀▀▀ █    
    ▀▀▀▀▀▀▀ █▄█▄▀▄▀ ▀ ▀▀▀▀▀▀▀    
    ▀▄█▀▀▀▀▄ ▄▄   ▀▀▄ ▀█▀▀▀▄     
     ▀█▀▄▄▀ ███▄▄ ▄▀▄█ █▀▀ ▀█    
    █▀▄▀██▀▄▀ █ ▄▄▄  ▀▀▄▀▄▀█▀    
    █  ▄▀█▀▄▀▀█ ▄█▀███ ██▀ ▀█    
    ▀ ▀ ▀▀▀ ███ ▀ ▀██▀▀▀█▄▀      
    █▀▀▀▀▀█ ▄  █▀ ▄ █ ▀ █▄▀▀▀    
    █ ███ █ █ ▄▀▄▄▄█▀██▀█▄█▄█    
    █ ▀▀▀ █ ▀▀  ▄█▀▄█▄▄▄▄█▀ █    
    ▀▀▀▀▀▀▀ ▀▀▀ ▀ ▀  ▀ ▀▀▀▀▀▀    
                                 
                                 "
`;
